scripts/Muon/GUI/Common/muon_load_data.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import (absolute_import, division, print_function)
+
+
+class MuonLoadData:
+    """
+    Lightweight 'struct' to store the results of loading from a load widget. Hard-code all required parameters for each
+    entry in the __init__ method below (this way, extending for new parameters is easy).
+
+    - Can be extended to add as many parameters as needed; all getting/setting is done through keyword arguments.
+    - Can be used as an iterator, with elements being dictionaries of parameter:value pairs.
+
+    The keywords in all methods are "greedy" and will match any entries, with an OR like behaviour for
+    multiple keywords. So for example "run=1234, filename ="file.nxs" would match to an entry with run=1234 or
+    filename = "file2.nxs".
+
+    - Clients responsibility to prevent duplicated entries.
+    - Clients responsibility to ensure the entries are correct (i.e. no validation is performed).
+
```

Comment:
add some useful (but not too long) links to introductory materials/tutorials on iterators